python/ctsm/site_and_regional/base_case.py

```diff
@@ -0,0 +1,184 @@
+"""
+This module includes the definition for a parent class for SinglePointCase
+and RegionalCase. The common functionalities of SinglePointCase and
+RegionalCase are defined in this Class.
+"""
+# -- Import libraries
+
+# -- standard libraries
+import os
+import logging
+from collections import namedtuple
+
+from datetime import date
+from getpass import getuser
+
+# -- 3rd party libraries
+import numpy as np
+import xarray as xr
+
+# -- import local classes for this script
+from ctsm.git_utils import get_git_short_hash
+
+USRDAT_DIR = "CLM_USRDAT_DIR"
+logger = logging.getLogger(__name__)
+
+# named tuple for datm input/output files and folder names
+DatmFiles = namedtuple(
+    "DatmFiles",
+    "indir outdir fdomain_in dir_solar dir_prec dir_tpqw tag_solar tag_prec tag_tpqw name_solar "
+    "name_prec name_tpqw "
+)
+
+class BaseCase:
+    """
+    Parent class to SinglePointCase and RegionalCase
+    ...
+    Attributes
+    ----------
+    create_domain : bool
+        flag for creating domain file
+    create_surfdata : bool
+        flag for creating surface dataset
+    create_landuse : bool
+        flag for creating landuse file
+    create_datm : bool
+        flag for creating DATM files
+    create_user_mods
+        flag for creating a user_mods directory
+    Methods
+    -------
+    create_1d_coord(filename, lon_varname , lat_varname,x_dim , y_dim )
+        create 1d coordinate variables to enable sel() method
+    add_tag_to_filename(filename, tag)
```

Comment:
- [ ] This is no longer a member of this class

Reply:
Thanks. Removed and I update the docstring in `add_tag_to_filename` in utils.py.

Reply:
GTG